index.js

```diff
@@ -604,15 +610,19 @@ module.exports = function (log, indexesPath) {
         status.batchUpdate(indexes, coreIndexNames)
         status.batchUpdate(newIndexes, newIndexNames)
 
-        waitingIndexCreate.get(waitingKey).forEach((cb) => cb())
-        waitingIndexCreate.delete(waitingKey)
+        runWaitingIndexLoadCbs(waitingIndexCreate, waitingKey)
 
         cb()
       },
     })
   }
 
+  // concurrent index load
+  const waitingIndexLoad = new Map()
+
   function loadLazyIndex(indexName, cb) {
+    if (onlyOneIndexAtATime(waitingIndexLoad, indexName, cb)) return
```

Comment:
I understand the intent with this code, but I'm confused whether it will actually work. Basically, we're putting this `cb` here in the Map, so it can be called later. And indeed in line 638 it will be called. But the `cb` in this case doesn't actually perform "index load tasks" such as lines 632–636, instead, the `cb` just informs that the loadLazyIndex is "done".